### PR TITLE
[Security] Use textContent instead innerHTML to remediate DOM based XSS

### PR DESCRIPTION
**Pull request by qazbnm456**
This commit is a security improvement for https://github.com/electron/electron/pull/10076, as @javan suggested, we should use `textContent` instead. Besides, I also revise some code to improve code quality.

Ref: https://www.owasp.org/index.php/DOM_based_XSS_Prevention_Cheat_Sheet

### CODE REVIEW

File: lib/renderer/content-scripts-injector.js [hunk 1]
```diff
@@ -23,12 +23,30 @@ const runContentScript = function (extensionId, url, code) {
   return compiledWrapper.call(this, context.chrome)
 }
 
+const runStylesheet = function (url, code) {
+  const wrapper = `(function (code) {
+    function init() {
+      var styleElement = document.createElement('style');
+      styleElement.setAttribute('type', 'text/css');
```
Comment (sindresorhus) [inline lib/renderer/content-scripts-injector.js]: You don't have to set this. It's the default.

File: lib/renderer/content-scripts-injector.js [hunk 2]
```diff
@@ -23,12 +23,30 @@ const runContentScript = function (extensionId, url, code) {
   return compiledWrapper.call(this, context.chrome)
 }
 
+const runStylesheet = function (url, code) {
+  const wrapper = `(function (code) {
+    function init() {
+      var styleElement = document.createElement('style');
```
Comment (sindresorhus) [inline lib/renderer/content-scripts-injector.js]: `var` => `const`

File: lib/renderer/content-scripts-injector.js [hunk 3]
```diff
@@ -23,12 +23,30 @@ const runContentScript = function (extensionId, url, code) {
   return compiledWrapper.call(this, context.chrome)
 }
 
+const runStylesheet = function (url, code) {
+  const wrapper = `(function (code) {
```
Comment (sindresorhus) [inline lib/renderer/content-scripts-injector.js]: `(code => {`

File: lib/renderer/content-scripts-injector.js [hunk 4]
```diff
@@ -23,12 +23,30 @@ const runContentScript = function (extensionId, url, code) {
...
+  return compiledWrapper.call(this, code);
+}
+
 // Run injected scripts.
 // https://developer.chrome.com/extensions/content_scripts
 const injectContentScript = function (extensionId, script) {
   if (!script.matches.some(matchesPattern)) return
 
-  if (script.js) {
+  if (script.js.length) {
```
Comment (MarshallOfSound) [inline lib/renderer/content-scripts-injector.js]: We want this to be `script.js`, if the length is 0 we don't care, the loop won't execute.

We need to check if the `js` property is defined at all

File: lib/renderer/content-scripts-injector.js [hunk 5]
```diff
@@ -41,13 +59,16 @@ const injectContentScript = function (extensionId, script) {
...
   }
 
-  if (script.css) {
-    for (const {code} of script.css) {
-      process.once('document-end', () => {
-        var node = document.createElement('style')
-        node.innerHTML = code
-        window.document.body.appendChild(node)
-      })
+  if (script.css.length) {
```
Comment (MarshallOfSound) [inline lib/renderer/content-scripts-injector.js]: Same as above 👍

File: lib/renderer/content-scripts-injector.js [hunk 6]
```diff
@@ -41,13 +59,16 @@ const injectContentScript = function (extensionId, script) {
...
+    for (const {url, code} of script.css) {
+      const fire = runStylesheet.bind(window, url, code)
+      if (script.runAt === 'document_start') {
+        process.once('document-start', fire)
+      } else if (script.runAt === 'document_end') {
+        process.once('document-end', fire)
+      } else if (script.runAt === 'document_idle') {
```
Comment (MarshallOfSound) [inline lib/renderer/content-scripts-injector.js]: This should just be an `else`, the spec says `document_idle` is the default 👍

**Comment by qazbnm456** [comment]
Thanks for your suggestions, @sindresorhus and @MarshallOfSound. I've made the changes already. 😄

**Pull request merged**